packages/talos_suffix/src/suffix.rs

```diff
@@ -18,11 +21,29 @@ impl<T> Suffix<T>
 where
     T: Sized + Clone + std::fmt::Debug,
 {
-    pub fn new(capacity: usize) -> Suffix<T> {
-        let suffix_vec: Vec<Option<SuffixItem<T>>> = vec![None; capacity + 1];
-        let messages = VecDeque::from(suffix_vec);
-
-        let meta = SuffixMeta { head: 0, prune_vers: None };
+    pub fn with_config(config: SuffixConfig) -> Suffix<T> {
+        let SuffixConfig {
+            capacity,
+            prune_start_threshold,
```

Comment:
I think we can add docs on core components. Like meaning and use case of new variables. It becomes easier to understand the change.

Reply:
Basic documentation added just for this method for now. Will add more documentation in subsequent PRs